eclipse-cla/src/main/java/org/eclipse/foundation/gerrit/validation/EclipseCommitValidationListener.java

```diff
@@ -124,70 +155,102 @@
+				messages.add(new CommitValidationMessage("The author has a current Eclipse Contributor Agreement (ECA) on file.", false));
 			} else {
-				messages.add(new CommitValidationMessage("The author does not have a current Eclipse Contributor Agreement (ECA) on file.\n" + 
-				"If there are multiple commits, please ensure that each author has a ECA.", true));	
+				messages.add(new CommitValidationMessage("The author does not have a current Eclipse Contributor Agreement (ECA) on file.\n" +
+				"If there are multiple commits, please ensure that each author has a ECA.", true));
 				addEmptyLine(messages);
 				errors.add("An Eclipse Contributor Agreement is required.");
 			}
-			
+
 			if (hasSignedOff(author, commit)) {
 				messages.add(new CommitValidationMessage("The author has \"signed-off\" on the contribution.", false));
 			} else {
 				messages.add(new CommitValidationMessage("The author has not \"signed-off\" on the contribution.\n" +
 					"If there are multiple commits, please ensure that each commit is signed-off.", true));
 				errors.add("The contributor must \"sign-off\" on the contribution.");
 			}
-			
+
 			// TODO Extend exception-throwing delegation to include all possible messages.
 			if (!errors.isEmpty()) {
 				addDocumentationPointerMessage(messages);
 				throw new CommitValidationException(errors.get(0), messages);
 			}
 		}
-		
+
 		addEmptyLine(messages);
-		
+
 		/*
 		 * Only committers can push on behalf of other users. Note that, I am asking
 		 * if the user (i.e. the person who is doing the actual push) is a committer.
 		 */
 		if (!author.getAccount().equals(user.getAccount())) {
 			if (!isCommitter(user, project)) {
-				messages.add(new CommitValidationMessage("You are not a project committer.", true));
-				messages.add(new CommitValidationMessage("Only project committers can push on behalf of others.", true));
-				addDocumentationPointerMessage(messages);
-				addEmptyLine(messages);
-				throw new CommitValidationException("You must be a committer to push on behalf of others.", messages);	
+				if (!getPreviousPatchsetAuthors(project, commit).contains(author)) {
+					messages.add(new CommitValidationMessage("You are not a project committer.", true));
+					messages.add(new CommitValidationMessage("Only project committers can push on behalf of others.", true));
+					addDocumentationPointerMessage(messages);
+					addEmptyLine(messages);
+					throw new CommitValidationException("You must be a committer to push on behalf of others or keep author of previous patchsets.",
+							messages);
+				}
 			}
-		} 
+		}
 
 		messages.add(new CommitValidationMessage("This commit passes Eclipse validation.", false));
-		
+
 		return messages;
 	}
 
+	private Set<IdentifiedUser> getPreviousPatchsetAuthors(Project project, RevCommit commit) {
+		Set<IdentifiedUser> authors = new HashSet<>();
+		try {
+			List<ChangeInfo> cis = gapi.changes().query(commit.getName()).get();
```

Comment:
I think there is a logic issue there. When testing, I'm getting java.lang.ArrayIndexOutOfBoundsException on the next line.

If I understand correctly, commit.getName() returns the sha of the commit being validated, so it's not in the repo yet and "cis" will then always be an empty list.

Shouldn't we use the changeId (if any) from the commit message to query the change api? In this case, when a non-committer creates changeset with an author!=user, the query will also return an empty list, correct?